src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -0,0 +1,141 @@
+                private route: ActivatedRoute) {
+    }
+
+    ngOnInit() {
+        this.subscription = this.route.params.subscribe(params => {
+            this.exerciseId = parseInt(params['exerciseId'], 10);
+            this.courseId = parseInt(params['courseId'], 10);
+        });
+
+        if (this.exercise === undefined) {
+            this.exerciseService.findResultsForExercise(this.exerciseId).subscribe((exercise: Exercise) => {
+                this.exercise = exercise;
+                this.exercise.isAtLeastTutor = this.accountService.isAtLeastTutorInCourse(this.exercise.course);
+                this.setExerciseStatusBadge();
+                if (this.hasResults) {
+                    this.sortedResults = this.exercise.participations[0].results.sort((a, b) => {
+                        const aValue = moment(a.completionDate).valueOf();
+                        const bValue = moment(b.completionDate).valueOf();
+                        return aValue - bValue;
+                    });
+                    const sortedResultLength = this.sortedResults.length;
+                    const startingElement = sortedResultLength - MAX_RESULT_HISTORY_LENGTH;
+                    this.sortedHistoryResult = this.sortedResults.slice(startingElement, sortedResultLength);
+                }
+            });
+        }
+    }
+
+    backToCourse() {
+        this.$location.back();
+    }
+
+    setExerciseStatusBadge(): void {
+        this.exerciseStatusBadge = moment(this.exercise.dueDate).isBefore(moment()) ? 'badge-danger' : 'badge-success';
+    }
+
+    exerciseRatedBadge(result: Result): string {
+        return result.rated ? 'badge-success' : 'badge-info';
+    }
+
+    get exerciseIsOver(): boolean {
+        return moment(this.exercise.dueDate).isBefore(moment());
+    }
+
+    get hasMoreResults(): boolean {
+        return this.sortedResults.length > MAX_RESULT_HISTORY_LENGTH;
+    }
+
+    get exerciseRouterLink(): string {
+        if (this.exercise.type === ExerciseType.MODELING) {
+            return `/course/${this.courseId}/exercise/${this.exercise.id}/assessment`;
+        } else if (this.exercise.type === ExerciseType.TEXT) {
+            return `/text/${this.exercise.id}/assessment`;
+
+        } else {
+            return;
+        }
+    }
+
+    get hasResults(): boolean {
+        const hasParticipations = this.exercise.participations && this.exercise.participations[0];
+        return hasParticipations && this.exercise.participations[0].results && this.exercise.participations[0].results.length > 0;
+    }
+
+    get currentResult(): Result {
+        if (!this.exercise.participations || !this.exercise.participations[0].results) {
+            return null;
+        }
+        const results = this.exercise.participations[0].results;
+        return results.filter(el => el.rated).pop();
+    }
+
+    get exerciseIcon(): ExerciseIcon {
+        if (this.exercise.type === this.PROGRAMMING) {
```

Comment:
I think it would be better to use a switch statement here.
Also: what about translations?

Reply:
translations can be done by `translateService: TranslateService`

if you prefer switch statements it can be changed of course